features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -0,0 +1,412 @@
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+class RumSessionEndedMetricTest {
+
+    @StringForgery
+    private lateinit var fakeSessionId: String
+
+    @Forgery
+    private lateinit var fakeStartReason: RumSessionScope.StartReason
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Test
+    fun `M create attributes with session ended metric type W toMetricAttributes`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        assertEquals(SessionEndedMetric.METRIC_TYPE_VALUE, attributes[SessionEndedMetric.METRIC_TYPE_KEY])
+    }
+
+    @Test
+    fun `M create attributes with session ended process type W toMetricAttributes`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val processType = rse[SessionEndedMetric.PROCESS_TYPE_KEY] as String
+        assertEquals(SessionEndedMetric.PROCESS_TYPE_VALUE, processType)
+    }
+
+    @Test
+    fun `M create attributes with session ended precondition W toMetricAttributes`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val precondition = rse[SessionEndedMetric.PRECONDITION_KEY] as String
+        assertEquals(fakeStartReason.asString, precondition)
+    }
+
+    @Test
+    fun `M init attributes W creating empty metric`() {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        val attributes = sessionEndedMetric.toMetricAttributes()
+
+        // Then
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+
+        assertEquals(rse[SessionEndedMetric.DURATION_KEY] as Long, 0L)
+
+        assertFalse(rse[SessionEndedMetric.WAS_STOPPED_KEY] as Boolean)
+
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY] as Int, 0)
+        assertEquals(viewCounts[SessionEndedMetric.VIEW_COUNTS_BG_KEY] as Int, 0)
+        assertEquals(viewCounts[SessionEndedMetric.VIEW_COUNTS_APP_LAUNCH_KEY] as Int, 0)
+
+        val errorCounts = rse[SessionEndedMetric.SDK_ERRORS_COUNT_KEY] as Map<*, *>
+        assertEquals(errorCounts[SessionEndedMetric.SDK_ERRORS_COUNT_TOTAL_KEY] as Int, 0)
+        Assertions.assertTrue((errorCounts[SessionEndedMetric.SDK_ERRORS_COUNT_BY_KIND_KEY] as Map<*, *>).isEmpty())
```

Comment:
static import to be used

```suggestion
        assertThat(errorCounts[SessionEndedMetric.SDK_ERRORS_COUNT_BY_KIND_KEY] as Map<*, *>).isEmpty()
```

bonus point for using `assertThat(...).isEmpty(...)` is more detailed message in case of failure, rather that something like "condition was false, but expected to to be true".

Reply:
Updated